src/mumble/AudioConfigDialog.cpp

```diff
@@ -511,21 +507,36 @@ void AudioInputDialog::updateEchoEnableState() {
 		outputInterface = s.qsAudioOutput;
 	}
 
-	if (air->canEcho(outputInterface)) {
+	qcbEcho->clear();
+	qcbEcho->setToolTip(QObject::tr("If enabled this tries to cancel out echo from the audio stream."));
+	qcbEcho->setCurrentIndex(0);
+
+	bool hasUsableEchoOption = false;
+
+	qcbEcho->insertItem(0, tr("Disabled"), "disabled");
+	qcbEcho->setItemData(0, tr("Disable echo cancellation."), Qt::ToolTipRole);
+
+	for (int i=0; i<air->echoOptions.count(); ++i) {
+		EchoCancellationOption eco = air->echoOptions[i];
```

Comment:
Would something like this also be possible instead?
```suggestion
	for (EchoCancellationOption eco : air->echoOptions) {
```

if so then this would be a more modern and robust way of writing such loops.

I the `insertItem` and `setItemData` could then be replaced by a `addItem` call

Reply:
Yeah. C++ trick. Will do.

Reply:
I checked the manual, and it seemed that these two function calls can't be combined because I need to set the `Qt::ToolTipRole`.